### PR TITLE
Adding hidden states restoring and burn in frames features to DRQN

### PR DESCRIPTION
**Pull request by hnekoeiq**

### CODE REVIEW

File: hive/agents/__init__.py [hunk 1]
```diff
@@ -13,6 +15,8 @@
         "LegalMovesRainbowAgent": LegalMovesRainbowAgent,
         "RainbowDQNAgent": RainbowDQNAgent,
         "RandomAgent": RandomAgent,
+        "DRQNAgent": DRQNAgent,
+        "DRQNhiddenAgent": DRQNhiddenAgent,
```
Comment (hnekoeiq) [inline hive/agents/__init__.py]: Remove this?

File: hive/agents/drqn.py [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+            and self._update_period_schedule.update()
+        ):
+            batch = self._replay_buffer.sample(batch_size=self._batch_size)
+            (
+                current_state_inputs,
+                next_state_inputs,
+                batch,
+            ) = self.preprocess_update_batch(batch)
+
+            if self._rnn_type == "lstm" and self._store_hidden == True:
```
Comment (hnekoeiq) [inline hive/agents/drqn.py]: Make this a nested if statement? We should take care of the case where rnn_type is nether lstm not gru.

File: hive/configs/atari/drqn_hidden.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
+run_name: &run_name 'atari-drqn-hidden-burn-frames'
+train_steps: 50000000
```
Comment (hnekoeiq) [inline hive/configs/atari/drqn_hidden.yml]: Just one yaml file should be enough

File: hive/replays/recurrent_hidden_replay.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import string
```
Comment (hnekoeiq) [inline hive/replays/recurrent_hidden_replay.py]: Remove this if `recurrent_replay.py` has everything.

File: hive/replays/recurrent_replay.py [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+            rnn_hidden_size (int): Size of the rnn hidden size.
+            store_hidden (bool): whether to store the hidden state or not.
+        """
+        if extra_storage_types is None:
+            extra_storage_types = {}
+        if rnn_type == "lstm" and store_hidden == True:
+            extra_storage_types["hidden_state"] = (np.float32, (1, 1, rnn_hidden_size))
+            extra_storage_types["cell_state"] = (np.float32, (1, 1, rnn_hidden_size))
+        if rnn_type == "gru" and store_hidden == True:
+            extra_storage_types["hidden_state"] = (np.float32, (1, 1, rnn_hidden_size))
```
Comment (hnekoeiq) [inline hive/replays/recurrent_replay.py]: again take care of the case where rnn_type is not valid. Through a value error for example.

File: hive/agents/drqn.py (hunk 2 above)
Comment (karthiks1701) [reply hive/agents/drqn.py]: Like in one `if` for `rnn_type` and the inner if for the hidden state?

**Comment by hnekoeiq** [comment]
Created a new PR. This one was out of date.

**Pull request closed without merging**